// @storybook/addon-actions/dist/components/ActionLogger/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactInspector = require('react-inspector');

var _reactInspector2 = _interopRequireDefault(_reactInspector);

var _emotionTheming = require('emotion-theming');

var _components = require('@storybook/components');

var _style = require('./style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ActionLogger = (0, _emotionTheming.withTheme)(function (_ref) {
  var actions = _ref.actions,
      onClear = _ref.onClear,
      theme = _ref.theme;
  return _react2.default.createElement(
    _style.Wrapper,
    null,
    _react2.default.createElement(
      _style.Actions,
      null,
      actions.map(function (action) {
        return _react2.default.createElement(
          _style.Action,
          { key: action.id },
          _react2.default.createElement(
            _style.Countwrap,
            null,
            action.count > 1 && _react2.default.createElement(
              _style.Counter,
              null,
              action.count
            )
          ),
          _react2.default.createElement(
            _style.InspectorContainer,
            null,
            _react2.default.createElement(_reactInspector2.default, {
              theme: theme.addonActionsTheme || 'chromeLight',
              sortObjectKeys: true,
              showNonenumerable: false,
              name: action.data.name,
              data: action.data.args || action.data
            })
          )
        );
      })
    ),
    _react2.default.createElement(
      _components.ActionBar,
      null,
      _react2.default.createElement(
        _components.ActionButton,
        { onClick: onClear },
        'CLEAR'
      )
    )
  );
});

ActionLogger.propTypes = {
  onClear: _propTypes2.default.func.isRequired,
  actions: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    count: _propTypes2.default.node,
    data: _propTypes2.default.shape({
      name: _propTypes2.default.node.isRequired,
      args: _propTypes2.default.any
    })
  })).isRequired
};

exports.default = ActionLogger;
