// @storybook/react-native/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getStorybookUI = exports.getStorybook = exports.configure = exports.clearDecorators = exports.addParameters = exports.addDecorator = exports.setAddon = exports.storiesOf = undefined;

var _preview = require('./preview');

var _preview2 = _interopRequireDefault(_preview);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var preview = new _preview2.default();

var storiesOf = exports.storiesOf = preview.storiesOf.bind(preview);
var setAddon = exports.setAddon = preview.setAddon.bind(preview);
var addDecorator = exports.addDecorator = preview.addDecorator.bind(preview);
var addParameters = exports.addParameters = preview.addParameters.bind(preview);
var clearDecorators = exports.clearDecorators = preview.clearDecorators.bind(preview);
var configure = exports.configure = preview.configure.bind(preview);
var getStorybook = exports.getStorybook = preview.getStorybook.bind(preview);
var getStorybookUI = exports.getStorybookUI = preview.getStorybookUI.bind(preview);
