// @storybook/core/dist/server/config/utils.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.nodePaths = exports.nodeModulesPaths = exports.excludePaths = exports.includePaths = undefined;

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

var _stringify = require('babel-runtime/core-js/json/stringify');

var _stringify2 = _interopRequireDefault(_stringify);

exports.loadEnv = loadEnv;
exports.getEntries = getEntries;

var _fs = require('fs');

var _fs2 = _interopRequireDefault(_fs);

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

var _nodeLogger = require('@storybook/node-logger');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var includePaths = exports.includePaths = [_path2.default.resolve('./')];

var excludePaths = exports.excludePaths = [_path2.default.resolve('node_modules')];

var nodeModulesPaths = exports.nodeModulesPaths = _path2.default.resolve('./node_modules');

var nodePaths = exports.nodePaths = (process.env.NODE_PATH || '').split(process.platform === 'win32' ? ';' : ':').filter(Boolean).map(function (p) {
  return _path2.default.resolve('./', p);
});

// Load environment variables starts with STORYBOOK_ to the client side.
function loadEnv() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

  var defaultNodeEnv = options.production ? 'production' : 'development';
  var env = {
    NODE_ENV: (0, _stringify2.default)(process.env.NODE_ENV || defaultNodeEnv),
    // This is to support CRA's public folder feature.
    // In production we set this to dot(.) to allow the browser to access these assests
    // even when deployed inside a subpath. (like in GitHub pages)
    // In development this is just empty as we always serves from the root.
    PUBLIC_URL: (0, _stringify2.default)(options.production ? '.' : '')
  };

  (0, _keys2.default)(process.env).filter(function (name) {
    return (/^STORYBOOK_/.test(name)
    );
  }).forEach(function (name) {
    env[name] = (0, _stringify2.default)(process.env[name]);
  });

  return {
    'process.env': env
  };
}

function getEntries(configDir) {
  var preview = [require.resolve('./polyfills'), require.resolve('./globals')];
  var manager = [require.resolve('./polyfills'), require.resolve('../../client/manager')];

  // Check whether a config.js file exists inside the storybook
  // config directory and throw an error if it's not.
  var storybookConfigPath = _path2.default.resolve(configDir, 'config.js');
  if (!_fs2.default.existsSync(storybookConfigPath)) {
    throw new Error('=> Create a storybook config file in "' + configDir + '/config.js".');
  }

  preview.push(require.resolve(storybookConfigPath));

  // Check whether addons.js file exists inside the storybook.
  var storybookCustomAddonsPath = _path2.default.resolve(configDir, 'addons.js');
  if (_fs2.default.existsSync(storybookCustomAddonsPath)) {
    _nodeLogger.logger.info('=> Loading custom addons config.');
    manager.unshift(storybookCustomAddonsPath);
  }

  return { preview: preview, manager: manager };
}
